File: src/com/rian/difficultycalculator/attributes/TimedDifficultyAttributes.java
package com.rian.difficultycalculator.attributes;

/**
 * Wraps a <code>DifficultyAttributes</code> object and adds a time value for which the attribute is valid.
 * <br><br>
 * Output by <code>DifficultyCalculator.calculateTimed</code> methods.
 */
public class TimedDifficultyAttributes {
    /**
     * The non-clock-adjusted time value at which the attributes take effect.
     */
    public final double time;

    /**
     * The attributes.
     */
    public final DifficultyAttributes attributes;

    public TimedDifficultyAttributes(double time, DifficultyAttributes attributes) {
        this.time = time;
        this.attributes = attributes;
    }

    public int compareTo(TimedDifficultyAttributes other) {
        return Double.compare(time, other.time);
    }
}

File: src/com/rian/osu/difficultycalculator/attributes/TimedDifficultyAttributes.kt
package com.rian.osu.difficultycalculator.attributes

import com.rian.osu.difficultycalculator.calculator.DifficultyCalculator

/**
 * Wraps a [DifficultyAttributes] object and adds a time value for which the attribute is valid.
 *
 * Output by [DifficultyCalculator.calculateTimed] methods.
 */
class TimedDifficultyAttributes(
    /**
     * The non-clock-adjusted time value at which the attributes take effect.
     */
    @JvmField
    val time: Double,

    /**
     * The attributes.
     */
    @JvmField
    val attributes: DifficultyAttributes
) {
    operator fun compareTo(other: TimedDifficultyAttributes) = time.compareTo(other.time)
}
